android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/login/LoginViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package co.nimblehq.ic.kmm.suv.android.ui.screens.login
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import co.nimblehq.ic.kmm.suv.domain.usecase.LogInUseCase
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class LoginViewModel(
+    private val logInUseCase: LogInUseCase
+): ViewModel() {
+
+    data class UiState(
+        val email: String = "",
+        val password: String = "",
+        val isLogInSuccess: Boolean = false,
+        val isLoading: Boolean = false,
+        val errorMessage: String? = null
+    )
```

Comment:
For Android, we define the states on different flows rather than combine in single data class, ex:

```
private val _loadingState = MutableStateFlow<Boolean>(false)
private val _errorState = MutableStateFlow<String>("")
```

or I'd suggest applying [sealed class/interface](https://kotlinlang.org/docs/sealed-classes.html) in Kotlin for different uiState:

```
sealed interface UiState {
    object Success: UiState
    data class Error(val message: String) : UiState
    object Loading : UiState
}
```

Reply:
@hoangnguyen92dn As I mentioned in the description, I followed [this view model](https://github.com/android/architecture-samples/blob/main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModel.kt)

However, applying `sealed class` would be a good approach! Let's me try.

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/92/commits/c3e5eee44a768ed400b77ebb78d1ae4ab69c1567

P.s: I decided to define states on different flows, so we can apply the same approach for all view models